table/block_based_table_reader.cc

```diff
@@ -2866,6 +2883,13 @@ Status BlockBasedTable::VerifyChecksumInBlocks(
         false /* decompress */, false /*maybe_compressed*/,
         UncompressionDict::GetEmptyDict(), rep_->persistent_cache_options);
     s = block_fetcher.ReadBlockContents();
+    if (!s.ok() && index_iter->key() == kPropertiesBlock) {
```

Comment:
A second chance is needed only if `s.IsCorruption()` is true. How about `s.IsCorruption() && index_iter->key() == kPropertiesBlock`?